### DISCUSSION TITLE
History Command

### DISCUSSION THREAD

**Discussion by masehitam** [isontheline/pro.webssh.net#992] [Ideas]
Hi, thanks for the great app,
is it possible to consider adding a history feature ?
This feature could provide users with a way to track their actions and interactions within the application, and adding command history to snippets will be great.

thank you..

**Comment by isontheline** [comment]
Hi @masehitam 👋

Many thanks for your feedback and reaching me out 🙏

> is it possible to consider adding a history feature ?

Does your request is related to this one?
https://github.com/isontheline/pro.webssh.net/issues/908

> adding command history to snippets will be great.

Could you tell me more about this statement please?

Have a nice day ☀️

**Reply by isontheline** [reply]
@masehitam May I up this thread?

**Comment by ThiagoLimahDev** [comment]
I believe that in this case it would help a lot for the "up" key to show the last commands typed, as eventually I need to change interface by interface and in this case I need to retype the entire command, this is quite annoying and I could just press "up" and that's it .

**Reply by isontheline** [reply]
History shared across servers so?

**Reply by ThiagoLimahDev** [reply]
On the same equipment. I use SSH access a lot on Huawei/Cisco devices, the fact that I can press "up" and navigate to the last commands would be excellent!

**Reply by isontheline** [reply]
@masehitam I will open another issue as this one is about user interaction within the app and not within the servers themself

**Comment by JohnCronk79** [comment]
Hello,

I wanted to give a second vote to having a ‘command history’ function.
Here are some thoughts to consider (I use this app on my iPad):
- have the ability to save a history of commands
- ability to choose via tab/radio button, three separate histories (one a history for current viewed connection, second to see history of all commands across all connections, and third a list of favorite commands that you can save so it’s persistent when you close the app).
- “view options” should have ability to filter duplicate commands, sort commands by ‘time’ so recent command is at top, ‘alphabetically’, ‘frequency of use’ so more frequently used command is at top
- Ability to paste and/or execute command by selecting it
- ability to view history in the left side of screen along with ‘inventory’ and ‘sessions’
- Might be better to view it on the right side of screen in a collapsible tab similar to the ‘inventory/sessions’.
- This feature could include or be merged with ‘snipits’. Perhaps have snippets shown on a forth viewing tab/radio button selection.


Thank you for your consideration on adding this feature
🇨🇦🍻

ps on iOS, the ‘help’ button which opens up a browser window to GitHub, does not open as a ‘session’. This means I can’t easily switch from my ssh session and making a comment here on GitHub. Missed opportunity as lots of room for session n right of screen.

**Reply by isontheline** [reply]
Hello @JohnCronk79 👋

Many thanks for your feedback 🙏

> ps on iOS, the ‘help’ button which opens up a browser window to GitHub, does not open as a ‘session’. This means I can’t easily switch from my ssh session and making a comment here on GitHub. Missed opportunity as lots of room for session n right of screen.

You're right the help screen lacks of opening inside Safari. I'm opening an issue to improve it.

> Thank you for your consideration on adding this feature

I will 😉

I have opened an issue to work on it : #1140
